File: problem-088/solution.go
package problem088

// division implements division without division, multiplication, and modulus operator.
func division(dividend, divisor int) int {
	sign := 1
	if (dividend < 0 && divisor > 0) || (dividend > 0 && divisor < 0) {
		sign = -1
	}

	dividend = abs(dividend)
	divisor = abs(divisor)

	quotient := 0
	for dividend > divisor {
		dividend -= divisor
		quotient++
	}

	return sign * quotient
}

func abs(n int) int {
	if n < 0 {
		n *= -1
	}
	return n
}

File: problem-088/solution.py
from typing import List, Tuple

def division(dividend: int, divisor: int) -> int:
    # determinse sign, will be negative only if one of them is negative.
    sign = -1 if ((dividend < 0)) ^ (divisor < 0) else 1

    dividend = abs(dividend)
    divisor = abs(divisor)

    # division is just repeated substraction.
    quotient = 0
    while dividend >= divisor:
        dividend -= divisor
        quotient += 1

    return sign * quotient


tests: List[Tuple[Tuple[int, int], int]] = [
    ((10, 3), 3),
    ((43, -8), -5),
]

for test in tests:
    assert division(test[0][0], test[0][1]) == test[1]